pkg/graveler/ref/commit_walker.go

```diff
@@ -68,34 +68,38 @@ func (w *CommitWalker) Err() error {
 	return w.err
 }
 
-func FindLowestCommonAncestor(ctx context.Context, getter CommitGetter, addressProvider ident.AddressProvider, repositoryID graveler.RepositoryID, left, right graveler.CommitID) (*graveler.Commit, error) {
-	discoveredSet := make(map[string]struct{})
+func FindLowestCommonAncestor(ctx context.Context, getter CommitGetter, addressProvider ident.AddressProvider, repositoryID graveler.RepositoryID, left, right graveler.CommitID) *graveler.Commit {
 	iterLeft := NewCommitWalker(ctx, getter, repositoryID, left)
+	// find the set of all ancestors of the left commit:
+	leftAncestors := make(map[string]bool)
```

Comment:
so we're keeping in memory a set of all of left's ancesstors? this is unbounded and potentially very large..